cleanlab_studio/internal/tlm/validation.py

```diff
@@ -0,0 +1,178 @@
+import os
+from typing import Union, Sequence, Any
+from cleanlab_studio.errors import ValidationError
+from cleanlab_studio.internal.constants import (
+    _VALID_TLM_MODELS,
+    TLM_MAX_TOKEN_RANGE,
+    TLM_NUM_CANDIDATE_RESPONSES_RANGE,
+    TLM_NUM_CONSISTENCY_SAMPLES_RANGE,
+)
+
+
+SKIP_VALIDATE_TLM_OPTIONS = os.environ.get("CLEANLAB_STUDIO_SKIP_VALIDATE_TLM_OPTIONS", False)
+
+
+def validate_tlm_prompt(prompt: Union[str, Sequence[str]]) -> None:
+    if isinstance(prompt, str):
+        return
+
+    elif isinstance(prompt, Sequence):
```

Comment:
imo error messages would be a little more helpful if they were more specific, such as pointing out the incorrect item type like below
```
    elif isinstance(prompt, Sequence):
        for item in prompt:
            if not isinstance(item, str):
                raise ValidationError(
                    f"Item {item} in prompt is of invalid type {type(item)}, all items in the prompt list must be of type str."
                )
```

Reply:
we would want to point out all invalid indexes for this type of validation, which is slightly less trivial (especially considering the code duplication you pointed out)

This would be a non-breaking future API change, so we will defer this.